lib/std.js

```diff
@@ -539,31 +539,60 @@ GeneratorFunction.Yield = function(message) {
 GeneratorFunction.Yield.prototype = new Error();
 GeneratorFunction.Yield.prototype.constructor = GeneratorFunction.Yield;
 
-function StdStorage() {
-    this.data = {};
-    this.length = 0;
+function StdAccessor(initialValue) {
+    var value = initialValue;
 
-    this.commit = function() {
-        this.length = Object.keys(data).length;
+    this.get = function() {
+        return value;
     };
-    this.key = function(idx) {
-        var keyName = Object.keys(data)[idx];
-        return data[keyName];
+
+    this.set = function(v) {
+        value = v;
     };
+}
+StdAccessor.isAccessor = function(obj) {
+    return obj instanceof StdAccessor;
+};
+
+function StdStorage() {
+    var data = new StdAccessor({});
+    var length = new StdAccessor(0);
+
+    function commit() {
+        length.set(Object.keys(data.get()).length);
+    }
+
     this.setItem = function(keyName, keyValue) {
-        data[keyName] = keyValue;
-        this.commit();
+        var d = data.get();
+        d[keyName] = keyValue;
+        data.set(d);
+        commit();
     };
+
     this.getItem = function(keyName) {
-        return data[keyName];
+        return data.get()[keyName];
     };
+
     this.removeItem = function(keyName) {
-        delete data[keyName];
-        this.commit();
+        var d = data.get();
+        delete d[keyName];
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

We detected that you are using `delete` to remove items from an array. If we use `delete`, it will delete the object property, but will not reindex the array or update its length. It leaves `undefined` holes in the array. Therefore, we recommend you to use `splice` to remove items from an array.

[Learn more](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/delete)